platforms/apl/overlay.conf

```diff
@@ -0,0 +1,7 @@
+CONFIG_APOLLOLAKE=y
+CONFIG_INTEL_DMIC=y
```

Comment:
Can you explain a little bit what is an overlay? Is just a fragment of config file that gets applied over default config?

Reply:
Exactly. It just applies one by one over existing Kconfig entries.
Let me just paste part of the build output logs:
`Parsing /home/aborisov/workspace/software_configuration_pub/zephyrproject/zephyr/Kconfig
Loaded configuration '/home/aborisov/workspace/software_configuration_pub/zephyrproject/zephyr/boards/xtensa/intel_adsp_cavs25/intel_adsp_cavs25_defconfig'
Merged configuration '/home/aborisov/workspace/software_configuration_pub/zephyrproject/zephyr/samples/subsys/audio/sof/prj.conf'
Merged configuration '/home/aborisov/workspace/software_configuration_pub/platforms/tgl/overlay.conf'
Configuration saved to '/home/aborisov/workspace/software_configuration_pub/zephyrproject/build/zephyr/.config'
Kconfig header saved to '/home/aborisov/workspace/software_configuration_pub/zephyrproject/build/zephyr/include/generated/autoconf.h'`

This is TGL build example. So it basically:
1. Gets zephyr board defconfig values
2. Then applies prj.conf values on top of it
3. Then applies our new overlay.conf on top of it
<etc... may be further overlay files provided that will just overwrite values that are repeated>
Final "summary" of Kconfig settings is written to autoconf.h header file.

Reply:
@aborisovich this is very good information. If you can add this  please in the commit message would be great!

Reply:
overlays are documented in https://docs.zephyrproject.org/latest/application/index.html